delivery-metrics/src/delivery_metrics_issue_model.py

```diff
@@ -0,0 +1,50 @@
+from delivery_metrics_model import DeliveryMetricsModel
+
+class DeliveryMetricsIssueModel(DeliveryMetricsModel):
+
+	def syncIssue(self, issue: dict) -> int:
+
+		# validation
+		if not isinstance(issue, dict):
+			return None
+
+		# get values needed for sql statement
+		guid = issue.get('guid')
+		title = issue.get('title')
+		t = issue.get('type')
+		points = issue.get('points') or 0
+		opened_date = self.formatDate(issue.get('opened_date'))
+		closed_date = self.formatDate(issue.get('closed_date'))
+		parent_guid = issue.get('parent_guid')
+		epic_id = issue.get('epic_id')
+		sprint_id = issue.get('sprint_id')
+		status = issue.get('status')
+		is_closed = issue.get('is_closed')
+		effective = self.getEffectiveDate()
+
+		# get cursor to keep open across transactions
+		cursor = self.cursor()
+
+		# insert into dimension table: issue
+		sql_dim = "insert into issue (guid, title, type, opened_date, closed_date, parent_issue_guid, epic_id) values (?, ?, ?, ?, ?, ?, ?) on conflict(guid) do update set (title, type, opened_date, closed_date, parent_issue_guid, epic_id, t_modified) = (?, ?, ?, ?, ?, ?, current_timestamp) returning id"
```

Comment:
I can appreciate the simplicity of using `on conflict(guid) do update` to upsert the issues in a single statement, but as someone who's been on the consuming end of a dimension table where every row was modified each run whether or not the values in a given row actually changed, it can be somewhat misleading. I know _technically_ we only care about the current value, but in practice it can be helpful to know how frequently those values are turning over.

Long term would love to discuss the pros and cons of a single upsert statement vs. two statements one to insert the new issues, another to update the existing records that have changed.

Reply:
Fair enough. There was no deep thought here. I was simply going for less code, and therefore used the upsert, but I'm not opposed to multiple statements.

Reply:
This is fixed. I replaced all the upserts with select and update statements. Rows will be updated on each run only if column values have changed. Please see latest commits.